### PR TITLE
Instant Search: fix button styling in Twenty Twenty One theme

### LINKED ISSUE #25407: Search: sort buttons inherit some styles from the theme
### Impacted plugin

Jetpack, Search

### Steps to Reproduce

1. Enable Instant Search on a test site with the Twenty Twenty One theme.
2. Enable dark mode for your search modal in the Search customization settings.

### A clear and concise description of what you expected to happen.

Sort options should look consistent and look like links.

### What actually happened

The sort links unexpectedly have a background color and a bigger font-size than expected.

Screen Shot 2022-08-08 at 16 07 38

### Browser

Mozilla Firefox

### Other information

We recently changed these from links to buttons as part of our accessibility efforts in https://github.com/Automattic/jetpack/pull/25163.

### Platform (Simple, Atomic, or both?)

_No response_

### Reproducibility

Consistent

### Severity

_No response_

### Available workarounds?

_No response_

### Workaround details

_No response_

### PR DESCRIPTION
**Pull request by bluefuton**
Fixes https://github.com/Automattic/jetpack/issues/25407.

Broken out from https://github.com/Automattic/jetpack/pull/25414 to specifically fix the sort button styling.

#### Changes proposed in this Pull Request:

Fix the display of sort buttons in Instant Search. Some themes (such as Twenty Twenty One) override our styles using very specific selectors.

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. On a site with a Jetpack Search plan, enable the Twenty Twenty One theme.
2. Open the Overlay by appending `?s=&category=uncategorized` to your site address.
3. Confirm the buttons look correct in both Light & Dark themes.
4. Toggle the light/dark themes from `/wp-admin/admin.php?page=jetpack-search-configure`.

### CODE REVIEW

File: projects/packages/search/changelog/fix-search-sort-button-styling [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: fixed
+
+Instant Search: fix sort button styling in Twenty Twenty One themeo
```
Comment (bluefuton) [inline projects/packages/search/changelog/fix-search-sort-button-styling]:
```suggestion
Instant Search: fix sort button styling in Twenty Twenty One theme
```

**Comment by kangzj** [comment]
@bluefuton thanks for proposing this. It works as described, however, could we possibly include the fix for clear button as well?

image

**Comment by bluefuton** [comment]
>  It works as described, however, could we possibly include the fix for clear button as well?

Sure 👍 I'll make a new mixin because adding all that `!important` to the existing one feels a bit scary.

**Review by kangzj — DISMISSED** [review]
Thanks for the quick fix :shipit: 

It seems we could make it to the weekly release 👍

File: projects/packages/search/changelog/fix-search-sort-button-styling [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: fixed
+
+Instant Search: fix sort button styling in Twenty Twenty One theme
```
Comment (bluefuton) [inline projects/packages/search/changelog/fix-search-sort-button-styling]:
```suggestion
Instant Search: fix button styling in Twenty Twenty One theme
```

**Review by kangzj — APPROVED** [review]

**Pull request merged**